### ISSUE TITLE
Are you using val set for parameter tuning?

### ISSUE THREAD

**Issue by jindongwang** [donlan2710/gRDA-Optimizer#15]
Hi, thanks for your interesting work. I see from your code (https://github.com/donlan2710/gRDA-Optimizer/blob/master/directional_pruning/ResNet50-ImageNet/main_newgRDA_print_sparse_save_model_lr0.1per140ep.py) that you use the whole training set for training, and the valid set (which is literally the test set) for validation. This is not proper as you are using the test set to tune hyperparameters. Could you please tell me the reason for this?

**Comment by skchao74** [comment]
I think it is fair to say we haven't considered a test set in this experiment. The goal was to showcase the generalization performance by fine-tuning the hyperparameters. Considering a test set sounds like an important next step indeed.

**Comment by jindongwang** [comment]
I see. Most of the cv papers also used the same setting: train on training set and valid/test on valid set since the test set for imagenet is not available.
